pip_compile/scripts/compiler.py

```diff
@@ -0,0 +1,105 @@
+import argparse
+import json
+import subprocess
+import tempfile
+from pathlib import Path
+
+BUILD_TAG = "build"
+
+SCRIPTS = Path(__file__).parent
+
+# Container Paths
+CONTAINER_REQUIREMENTS_PATH = "/.pip-requirements/mount/"
+CONTAINER_SCRIPTS_PATH = "/.pip-requirements/scripts/"
+
+
+def build_image(image, tilt_resource, target):
+    dockerimages_json = subprocess.run(
+        ["tilt", "get", "dockerimages", f"{tilt_resource}:{image}", "-o", "json"],
+        capture_output=True,
+        text=True,
+    )
+    dockerimages = json.loads(dockerimages_json.stdout)
+    spec = dockerimages["spec"]
+
+    # with tempfile.NamedTemporaryFile(mode="w+") as dockerfile:
+    #     # Create temporary Dockerfile.
+    #     dockerfile.write(spec["dockerfileContents"])
+    #     dockerfile.close()
+    #
+    #     print(dockerfile.name)
+
+    docker_build = [
+        "docker",
+        "build",
+        "--quiet",
+        f"--tag={image}:{BUILD_TAG}",
+        f"--target={target}",
+        f"--cache-from={image}",
+        "--file=-",
+        spec["context"],
+    ]
+    process = subprocess.run(
+        docker_build,
+        input=spec["dockerfileContents"].encode("utf-8"),
+    )
+
+    if process.returncode != 0:
+        print("Error: Docker image failed to build")
+        print(process.stdout, process.stderr)
+        exit(1)
+
+
+def run_container(image, compile_args, local_req_path):
```

Comment:
Can you add type hints and docstring to this function?

Reply:
Yes, not sure how I missed this one!